samza-api/src/main/java/org/apache/samza/system/SystemAdmin.java

```diff
@@ -164,4 +165,10 @@ default void deleteMessages(Map<SystemStreamPartition, String> offsets) {
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Builds and returns the {@link StartpointVisitor} for a system.
+   */
+  default StartpointVisitor getStartpointVisitor() {
```

Comment:
Are you going to integrate this with the `OffsetManager` in a follow PR? I don't see it being used in the current PR.

Reply:
Would it be clearer to add a method called `resolveStartpoint`? Then the `SystemAdmin` impl can internally choose how to implement.